crates/cli/src/utils/mod.rs

```diff
@@ -623,6 +650,20 @@ mod tests {
         assert!(!p.is_sol_test());
     }
 
+    #[test]
+    fn can_parse_u256s_correctly() {
+        let num = "1";
+        let res = parse_u256(num);
+        assert!(res.is_ok());
+        let num = "0x10";
+        let res = parse_u256(num);
+        assert_eq!(U256::from(16), res.unwrap());
+        // This should not be parse-able, as we explicitly need prefixed hex strings.
```

Comment:
pushed some changes—the relevant part is removing the ability to implicitly parse hex strings—we don't want to do this, and even if users might feel some pain due to using unprefixed PKs, it's not recommended as it could be dangerous.